cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/ExecutionTaskPlanner.java

```diff
@@ -17,49 +17,64 @@
  * <ul>
- * <li>All proposals of leadership movements are put into the same list and will be executed together.
- * <li>Proposals of partition movements are tracked by broker and execution Id.
+ * <li>All leadership movement tasks are put into the same list and will be executed together.
+ * <li>Partition movements tasks are tracked by broker and execution Id.
  * </ul>
  * <p>
- * This class tracks the execution proposals for each broker using a sorted Set. The proposal's position in this set
- * represents its execution order and is determined by the passed in {@link ReplicaMovementStrategy} or
- * {@link #_defaultReplicaMovementTaskStrategy}. The proposal is tracked both under source broker and
- * destination broker's plan. Once a proposal is fulfilled, the proposal will be removed from both source broker and destination
- * broker's execution plan.
+ * This class tracks the partition movements for each broker using a sorted Set.
+ * The task's position in this set represents its execution order. For inter-broker partition movement task, the position
+ * is determined by the passed in {@link ReplicaMovementStrategy} or {@link ExecutionTaskPlanner#_defaultReplicaMovementTaskStrategy}.
+ * For intra-broker partition movement task, the position is determined by assigned execution id.
+ * The task is tracked both under source broker and destination broker's plan.
+ * Once a task is fulfilled, the task will be removed from both source broker and destination broker's execution plan.
  * <p>
  * This class is not thread safe.
  */
 public class ExecutionTaskPlanner {
   private final Logger LOG = LoggerFactory.getLogger(ExecutionTaskPlanner.class);
-  private Map<Integer, SortedSet<ExecutionTask>> _partMoveTaskByBrokerId;
-  private long _remainingDataToMove;
-  private final Set<ExecutionTask> _remainingReplicaMovements;
+  private Map<Integer, SortedSet<ExecutionTask>> _interPartMoveTaskByBrokerId;
+  private Map<Integer, SortedSet<ExecutionTask>> _intraPartMoveTaskByBrokerId;
+  private final Set<ExecutionTask> _remainingInterBrokerReplicaMovements;
+  private final Set<ExecutionTask> _remainingIntraBrokerReplicaMovements;
   private final Map<Long, ExecutionTask> _remainingLeadershipMovements;
   private long _executionId;
   private ReplicaMovementStrategy _defaultReplicaMovementTaskStrategy;
+  private final AdminClient _adminClient;
+  private final long LOGDIR_RESPONSE_TIMEOUT_MS = 1000L;
 
-  public ExecutionTaskPlanner(List<String> defaultReplicaMovementStrategies) {
+  public ExecutionTaskPlanner(AdminClient adminClient, List<String> defaultReplicaMovementStrategies) {
     _executionId = 0L;
-    _partMoveTaskByBrokerId = new HashMap<>();
-    _remainingReplicaMovements = new TreeSet<>();
-    _remainingDataToMove = 0L;
+    _interPartMoveTaskByBrokerId = new HashMap<>();
+    _intraPartMoveTaskByBrokerId = new HashMap<>();
+    _remainingInterBrokerReplicaMovements = new HashSet<>();
```

Comment:
Yes, since the execution order for inter-broker movement task is determined by [`ReplicaMovementStrategy`](https://github.com/linkedin/cruise-control/blob/d35af1b6d5a87046e6cd173948755a1e50faa531/cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/ExecutionTaskPlanner.java#L131) and for intra-broker is determined by Execution_ID. 
the order is actually maintained by SortedSet(notice `_interPartMoveTaskByBrokerId` and `_intraPartMoveTaskByBrokerId` are of type Map<Integer, SortedSet<ExecutionTask>> )